packages/hoppscotch-backend/src/team-collection/team-collection.service.ts

```diff
@@ -1056,4 +1063,248 @@ export class TeamCollectionService {
       return E.left(TEAM_COLL_NOT_FOUND);
     }
   }
+
+  /**
+   * Search for TeamCollections and TeamRequests by title
+   *
+   * @param searchQuery The search query
+   * @param take Number of items we want returned
+   * @param skip Number of items we want to skip
+   * @returns An Either of the search results
+   */
+  async searchByTitle(searchQuery: string, take = 10, skip = 0) {
+    // Fetch all collections and requests that match the search query
+    const searchResults: SearchQueryReturnType[] = [];
+
+    const matchedCollections = await this.searchCollections(
+      searchQuery,
+      take,
+      skip,
+    );
+    if (E.isLeft(matchedCollections))
+      return E.left(<AuthError>{
+        message: matchedCollections.left,
+        statusCode: HttpStatus.NOT_FOUND,
+      });
+    searchResults.push(...matchedCollections.right);
+
+    const matchedRequests = await this.searchRequests(searchQuery, take, skip);
+    if (E.isLeft(matchedRequests))
+      return E.left(<AuthError>{
+        message: matchedRequests.left,
+        statusCode: HttpStatus.NOT_FOUND,
+      });
+    searchResults.push(...matchedRequests.right);
+
+    // Generate the parent tree for searchResults
+    const searchResultsWithTree: CollectionSearchNode[] = [];
+
+    for (let i = 0; i < searchResults.length; i++) {
+      const fetchedParentTree = await this.fetchParentTree(searchResults[i]);
+      if (E.isLeft(fetchedParentTree))
+        return E.left(<AuthError>{
+          message: fetchedParentTree.left,
+          statusCode: HttpStatus.NOT_FOUND,
+        });
+      searchResultsWithTree.push({
+        type: searchResults[i].type,
+        title: searchResults[i].title,
+        method: searchResults[i].method,
+        id: searchResults[i].id,
+        path: !fetchedParentTree
+          ? []
+          : ([fetchedParentTree.right] as CollectionSearchNode[]),
+      });
+    }
+
+    return E.right({ data: searchResultsWithTree });
+  }
+
+  /**
+   * Search for TeamCollections by title
+   *
+   * @param searchQuery The search query
+   * @param take Number of items we want returned
+   * @param skip Number of items we want to skip
+   * @returns An Either of the search results
+   */
+  private async searchCollections(
+    searchQuery: string,
+    take: number,
+    skip: number,
+  ) {
+    const query = Prisma.sql`
+    select id,title,'collection' AS type
+    from "TeamCollection"
+    where titlesearch @@ to_tsquery(${searchQuery})
+    order by ts_rank(titlesearch,to_tsquery(${searchQuery}))
+    limit ${take}
+    OFFSET ${skip === 0 ? 0 : (skip - 1) * take};
+  `;
```

Comment:
In this query, we missed to add another where condition on `teamID`.

The consequence is,
Assume, there are 2 teams. One is `Balu's Team`, another `Arif's Team`
balu@hopp.io is the only member of `Balu's Team`. arif@hopp.io is the only member of `Arif's Team`.

Now, when Balu goes to search, he gets suggestions/responses from 'Arif's Team' as well.

Reply:
Understood, I'll push a fix for that ASAP.